### PR TITLE
feat: create and delete buckets based on namespace changes

### PR DESCRIPTION
**Pull request by 0xff-dev**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it
chore: create and delete buckets based on namespace changes

#### Which issue(s) this PR fixes

Fixes #

### CODE REVIEW

File: controllers/sync_namespace.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+	"github.com/kubeagi/arcadia/pkg/config"
+	"github.com/kubeagi/arcadia/pkg/datasource"
+	"github.com/kubeagi/arcadia/pkg/utils"
+)
+
+const (
+	BucketNotEmpty         = "The bucket you tried to delete is not empty"
+	BucketNotExist         = "The specified bucket does not exist"
+	SkipNamespaceConfigMap = "skip-namespaces"
```
Comment (nkwangleiGIT) [inline controllers/sync_namespace.go:42]: why we have to hold this skip-namespaces here?
pls add some comment for the design.

File: controllers/sync_namespace.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+		klog.Infof("namespace %s is in the filter list and will not be created, delete the corresponding bucket.", instance.Name)
+		return reconcile.Result{}, nil
+	}
+
+	if instance.DeletionTimestamp != nil {
+		err = r.removeBucket(ctx, instance.Name)
+		return reconcile.Result{RequeueAfter: 3 * time.Second}, err
+	}
+
+	err = r.syncBucket(ctx, instance.Name)
```
Comment (nkwangleiGIT) [inline controllers/sync_namespace.go:86]: so all events from namespace will invoke syncBucket method? is there any attribute( annotation etc..) that can be used to limit the rate of syncBucket, it's better to invoke once when a specific event occurs.

File: controllers/sync_namespace.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+	}
+	endpoint := systemDatasource.Spec.Enpoint.DeepCopy()
+	if endpoint.AuthSecret != nil && endpoint.AuthSecret.Namespace == nil {
+		endpoint.AuthSecret.WithNameSpace(systemDatasource.Namespace)
+	}
+	oss, err := datasource.NewOSS(ctx, r.Client, endpoint)
+	if err != nil {
+		klog.Errorf("generate new minio client error %s", err)
+		return err
+	}
```
Comment (nkwangleiGIT) [inline controllers/sync_namespace.go:141]: line 128 - 141 seems duplicated with 98-111

File: controllers/sync_namespace.go (hunk 2 above)
Comment (0xff-dev) [reply controllers/sync_namespace.go:86]: Maybe the `create` and `delete` actions do the processing just fine, and the `update` operation, can just be ignored.

File: controllers/sync_namespace.go [hunk 4]
```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2023 KubeAGI.
+
```
Comment (bjwswang) [inline controllers/sync_namespace.go:3]: how about `namespace_controller.go`?

File: controllers/namespace_controller.go [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+}
+
+func (r *NamespacetReconciler) removeBucket(ctx context.Context, namespace string) error {
+	oss, err := r.ossClient(ctx)
+	if err != nil {
+		err = fmt.Errorf("remove bucket: failed to get oss client error %w", err)
+		klog.Error(err)
+		return err
+	}
+	err = oss.Client.RemoveBucket(ctx, namespace)
```
Comment (nkwangleiGIT) [inline controllers/namespace_controller.go:152]: add a log when really remove bucket like create bucket at the next commit

**Pull request merged**